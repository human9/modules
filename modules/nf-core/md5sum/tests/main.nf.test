nextflow_process {

    name "Test Process MD5SUM"
    script "../main.nf"
    process "MD5SUM"

    tag "modules"
    tag "modules_nfcore"
    tag "md5sum"

    test("md5sum on hello.txt") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'hello' ],
                    [ file(params.modules_testdata_base_path + 'generic/txt/hello.txt', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            with(process.out) {
                // check that one output is ready on the channel
                assert checksum.size() == 1
                // destructure tuple
                (meta, md5file) = checksum.get(0)
                // ensure file is named as expected
                assert md5file ==~ ".*/hello.txt.md5"
                // compare checksum file content
                assert path(md5file).text == 'e59ff97941044f85df5297e1c302d260  hello.txt\n'

                // create / test snapshot
                assert snapshot(checksum).match("hello_checksum")
            }
        }
    }

    test("md5sum on hello.txt (BSD-style)") {

        config './nextflow.config'

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'hello' ],
                    [ file(params.modules_testdata_base_path + 'generic/txt/hello.txt', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            with(process.out) {
                // check that one output is ready on the channel
                assert checksum.size() == 1
                // destructure tuple
                (meta, md5file) = checksum.get(0)
                // ensure file is named as expected
                assert md5file ==~ ".*/hello.txt.md5"
                // compare checksum file content
                assert path(md5file).text == 'MD5 (hello.txt) = e59ff97941044f85df5297e1c302d260\n'

                // create / test snapshot
                assert snapshot(checksum).match("hello_checksum_BSD")
            }
        }
    }
    
}
