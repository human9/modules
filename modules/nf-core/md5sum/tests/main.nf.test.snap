{
    "hello_checksum": {
        "content": [
            [
                [
                    {
                        "id": "hello"
                    },
                    "hello.txt.md5:md5,5c18e1db5460fb32fed66966483165fd"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T21:37:57.984016644"
    },
    "hello_checksum_BSD": {
        "content": [
            [
                [
                    {
                        "id": "hello"
                    },
                    "hello.txt.md5:md5,152a03f5dc7aa8db6612f63154ecbca2"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-19T21:38:00.653988269"
    }
}